### PR TITLE
Use jakarta

### PR DESCRIPTION
**Pull request by brinxmat**
- Uses an xjc gradle plugin that has been updated recently (previous is basically dead)
- Does as much as possible in groovy, rather than via the plugin, which hopefully means we can drop-in replacement since no special functionality is required
- jakarta is now used

### CODE REVIEW

File: datacite-transform/build.gradle [hunk 1]
```diff
@@ -1,41 +1,80 @@
-import java.nio.file.Files
-import java.nio.file.Paths
-
 plugins {
     id 'java'
-    id "org.unbroken-dome.xjc" version "2.0.0"
```
Comment (brinxmat) [inline datacite-transform/build.gradle:6]: This was last updated four years ago, a PR was created by a third party for jakarta update in recent time, but nothing has been commented or merged in four years.

File: datacite-transform/build.gradle [hunk 2]
```diff
@@ -1,41 +1,80 @@
...
+    id("io.github.alexpercont.xjc-plugin") version "1.1"
```
Comment (brinxmat) [inline datacite-transform/build.gradle]: This is not necessarily the best choice, but it is the most recent and supports jakarta.

The download plugin is a common choice.

File: datacite-transform/build.gradle [hunk 3]
```diff
@@ -1,41 +1,80 @@
...
+    def xml = new groovy.xml.XmlSlurper().parseText(source.getText())
 
-    if (!file(schemaDir + urlFile).exists()) {
-        Files.createFile(Paths.get(schemaDir + urlFile))
-                .text = "https://schema.datacite.org/meta/kernel-${schemaVersion}/metadata.xsd"
+    return xml.'**'.findAll {
+        it.name() == 'include' || it.name() == 'import'
+    }.collect {
+        it.@'schemaLocation'
     }
```
Comment (brinxmat) [inline datacite-transform/build.gradle:40]: Here, we're extracting the URIs of the XML files that we need to gather in order to be able to build.

File: datacite-transform/build.gradle [hunk 4]
```diff
@@ -1,41 +1,80 @@
...
 }
 
+private File downloadFile(String path) {
+    def uri = "https://schema.datacite.org/meta/kernel-${SCHEMA_VERSION}/${path}"
+    def file = file(SCHEMA_DIR.toURI().toString() + '/' + path)
+
+    if (!file.exists()) {
```
Comment (brinxmat) [inline datacite-transform/build.gradle:47]: We don't need to download if we already have.

File: datacite-transform/build.gradle [hunk 5]
```diff
@@ -1,41 +1,80 @@
...
+        def destination = path.contains("/") ? file.parentFile : SCHEMA_DIR
+        destination.mkdirs()
+        download.run {
+            src uri
+            dest destination
+        }
+        DID_DOWNLOAD.put(path, true)
+    } else {
+        DID_DOWNLOAD.put(path, false)
```
Comment (brinxmat) [inline datacite-transform/build.gradle:56]: If we don't, we give the message that you might want to clean the source to override.

File: datacite-transform/build.gradle [hunk 6]
```diff
@@ -1,41 +1,80 @@
...
+    }
+    return file;
+}
+
 jar {
-    from "$buildDir/classes/java/main"
+    from "${layout.buildDirectory}/classes/java/main"
+}
+
+tasks.named('sourcesJar') {
```
Comment (brinxmat) [inline datacite-transform/build.gradle:65]: Necessary to provide explicit dependency here.

File: datacite-transform/build.gradle [hunk 7]
```diff
@@ -1,41 +1,80 @@
...
+    dependsOn('xjc')
 }
 
-System.setProperty('javax.xml.accessExternalSchema', 'all')
```
Comment (brinxmat) [inline datacite-transform/build.gradle:36]: This is what we cannot do

File: datacite-transform/src/main/java/no/unit/nva/transformer/Transformer.java [hunk 8]
```diff
@@ -1,11 +1,11 @@
 package no.unit.nva.transformer;
 
+import jakarta.xml.bind.JAXBContext;
+import jakarta.xml.bind.JAXBException;
+import jakarta.xml.bind.Marshaller;
```
Comment (brinxmat) [inline datacite-transform/src/main/java/no/unit/nva/transformer/Transformer.java:5]: Hurra

**Comment by ketilaa** [comment]
Have we considered just downloading the schema files? It is somewhat risky to rely on external content during a build, and a given build might be hard to reproduce: https://reproducible-builds.org

File: build.gradle [hunk 9]
```diff
@@ -16,18 +16,19 @@ allprojects {
     apply plugin: 'nebula.lint'
 
     group 'no.unit.nva'
-    version '0.5.8'
+    version '0.5.9'
 
-    sourceCompatibility = JavaVersion.VERSION_11
-    targetCompatibility = JavaVersion.VERSION_11
```
Comment (torbjokv) [inline build.gradle:22]: :-o

File: datacite-transform/build.gradle (hunk 2 above)
Comment (torbjokv) [reply datacite-transform/build.gradle]: I think @ketilaa have a point regarding the xsd and if we also don't need to rely on a plugin with short track record and only one contributor, that would be nice. If download is not a requirement, would it be easier to find an alternative?

**Review by LarsV123 — APPROVED** [review]

**Review by torbjokv — APPROVED** [review]

**Review by ketilaa — APPROVED** [review]

File: datacite-transform/build.gradle (hunk 2 above)
Comment (brinxmat) [reply datacite-transform/build.gradle]: We changed the plugin to another one-person (with two personalities) project, but with *more users*

**Comment by brinxmat** [comment]
> Have we considered just downloading the schema files? It is somewhat risky to rely on external content during a build, and a given build might be hard to reproduce: https://reproducible-builds.org

@ketilaa @torbjokv @LarsV123 I will create a new PR that makes the "download" functionality a "helper" and put the XSD under version control IF AND WHEN I know the licensing status of the XSD.

**Pull request merged**